Content.Server/Weapon/Melee/ZombieHand/ZombieHandSystem.cs

```diff
@@ -0,0 +1,23 @@
+using Content.Server.Disease;
+using Content.Server.Disease.Components;
+
+namespace Content.Server.Weapon.Melee.ZombieHand
+{
+    public sealed class ZombieHandSystem : EntitySystem
+    {
+        [Dependency] private readonly DiseaseZombieSystem _diseaseZombie = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ZombieHandComponent, MeleeHitEvent>(OnMeleeHit);
+        }
+
+        private void OnMeleeHit(EntityUid uid, ZombieHandComponent component, MeleeHitEvent args)
+        {
+            if(HasComp<DiseaseZombieComponent>(args.User))
+            {
+                _diseaseZombie.OnMeleeHit(uid, EntityManager.GetComponent<DiseaseZombieComponent>(args.User), args);
```

Comment:
uhhhhhhhhhhhhhhhhhhhhhhh this is kind of weird, why not just do the logic in here

Reply:
The reason it's implemented like this is that there's two distinct scenarios when infection needs to happen. Both when a creature with a hand attacks with with the zombie claw entity (in which the code above runs) but also if a creature without a hand (goat/penguin or something) attacks another entity, the logic still needs to run. I could probably rewrite it in some form so that it is contained on this, but it made more sense for the logic to be on the zombie component so that it can do all the bloodstream interfacing. 

It's up to you though. That component was just meant to be a simple patchwork to get the zombie claw item to work properly. The general implementation in the diseaseZombieSystem just makes it work for entities without hands.

Reply:
nvm that bs i'll just fix it.